src/controllers/PlaywrightController.ts

```diff
@@ -1,25 +1,36 @@
-import playwright from 'playwright-core';
+import playwright from 'playwright-chromium';
+import { parse } from 'url';
+import { getOrigin, tryJsonParse } from '../utils';
 import {
     BrowserController,
-    BrowserRequestHandler,
-    ResponseData,
     BrowserRequest,
+    BrowserRequestHandler,
     BrowserRequestType,
+    ResponseData,
 } from './BrowserController';
-import { getOrigin, tryJsonParse } from '../utils';
-import { parse } from 'url';
 
 export class PlaywrightController implements BrowserController {
+    private requestHandler?: (request: playwright.Route) => void;
+
     constructor(private readonly page: playwright.Page) {}
 
     async startInterception(onRequest: BrowserRequestHandler) {
-        this.page.route('**/*', (route: playwright.Route) => {
+        await this.stopInterception();
+        this.requestHandler = (route: playwright.Route) => {
```

Comment:
Ok. And what do you think about this version:
```ts
    private onRequest: BrowserRequestHandler = () => {};
    private hasInterceptingEnabled: boolean = false;

    constructor(private readonly page: playwright.Page) {}

    requestHandler = (route: playwright.Route) => {
        this.onRequest(
            this.toBrowserRequest(route.request()),
            (data) => this.respond(route, data),
            () => this.skip(route)
        );
    };

    async startInterception(onRequest: BrowserRequestHandler) {
        if (this.hasInterceptingEnabled === false) {
            this.hasInterceptingEnabled = true;
            this.onRequest = onRequest;
            await this.page.route('**/*', this.requestHandler);
        }
    }

    async stopInterception() {
        if (this.hasInterceptingEnabled) {
            await this.page.unroute('**/*', this.requestHandler);
            this.hasInterceptingEnabled = false;
        }
    }
```
?

and what if onRequest will change (theoretically) ?

Reply:
Now I see where the need of `this.requestHandler` was coming from - we needed somehow capture the reference to `onRequest` callback.

As you said, onRequest changes only theoretically - in practice it does change at all, because it is always passed`Mockiavelli.onRequest` method . So simplify things even further it could be passed to the constructor (through BrowserControllerFactory.createForPage). Then start and stop interception methods would has no args.

Reply:
So, what do you think about that one:
```typescript
    constructor(private readonly page: playwright.Page, private readonly onRequest: BrowserRequestHandler = () => {}) {}

    requestHandler = (route: playwright.Route) => {
        this.onRequest(
            this.toBrowserRequest(route.request()),
            (data) => this.respond(route, data),
            () => this.skip(route)
        );
    };

    async startInterception() {
           await this.page.route('**/*', this.requestHandler);
    }

    async stopInterception() {
            await this.page.unroute('**/*', this.requestHandler);
        }
    }
```
?

Reply:
👍